reactivesocket-tck-drivers/src/main/java/io/reactivesocket/tckdrivers/client/JavaClientDriver.java

```diff
@@ -0,0 +1,554 @@
+ */
+
+package io.reactivesocket.tckdrivers.client;
+
+import io.reactivesocket.Payload;
+import io.reactivesocket.ReactiveSocket;
+import io.reactivesocket.tckdrivers.common.*;
+import io.reactivesocket.util.PayloadImpl;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * This class is the driver for the Java ReactiveSocket client. To use with class with the current Java impl of
+ * ReactiveSocket, one should supply both a test file as well as a function that can generate ReactiveSockets on demand.
+ * This driver will then parse through the test file, and for each test, it will run them on their own thread and print
+ * out the results.
+ */
+public class JavaClientDriver {
+
+    // colors for printing things out
+    private final String ANSI_RESET = "\u001B[0m";
+    private final String ANSI_RED = "\u001B[31m";
+    private final String ANSI_GREEN = "\u001B[32m";
+
+    private final BufferedReader reader;
+    private final Map<String, TestSubscriber<Payload>> payloadSubscribers;
+    private final Map<String, TestSubscriber<Void>> fnfSubscribers;
+    private final Map<String, String> idToType;
+    private final Supplier<ReactiveSocket> createClient;
+
+    public JavaClientDriver(String path, Supplier<ReactiveSocket> createClient) throws FileNotFoundException {
+        this.reader = new BufferedReader(new FileReader(path));
+        this.payloadSubscribers = new HashMap<>();
+        this.fnfSubscribers = new HashMap<>();
+        this.idToType = new HashMap<>();
+        this.createClient = createClient;
+    }
+
+    /**
+     * Splits the test file into individual tests, and then run each of them on their own thread.
+     * @throws IOException
+     */
+    public void runTests() throws IOException {
+        List<List<String>> tests = new ArrayList<>();
+        List<String> test = new ArrayList<>();
+        String line = reader.readLine();
+        while (line != null) {
+            switch (line) {
+                case "!":
+                    tests.add(test);
+                    test = new ArrayList<>();
+                    break;
+                default:
+                    test.add(line);
+                    break;
+            }
+            line = reader.readLine();
+        }
+        tests.add(test);
+        tests = tests.subList(1, tests.size()); // remove the first list, which is empty
+        for (List<String> t : tests) {
+            TestThread thread = new TestThread(t);
+            thread.start();
+            thread.join();
```

Comment:
My idea is that if there are any exceptions, they would be contained within the thread and would allow subsequent tests to run.